### DISCUSSION TITLE
How to use to generate an open-api server

### DISCUSSION THREAD

**Discussion by venkatd** [aj-foster/open-api-generator#54] [Q&A] [answered]
Hi thanks this library looks great!

I am trying to define an open api schema so that I can generate both a frontend client (there is other tooling I'm using for this) and an Elixir server. I would want the server to define the proper endpoints, handle casting, etc.

It looks like, at the moment, this library is designed for developing Elixir client SDKs for existing APIs.

Is this a good library to extend for this usecase or will I be setting myself up for trouble?

**Comment by aj-foster** [comment, marked as answer]
Hi @venkatd,

You're correct that this library is designed for developing clients for existing APIs. Developing servers is an interesting problem, and this project probably isn't a good place for it. You should definitely feel free to use any of the reader code, but I think the rest of the code would cause more problems than solutions.

**Comment by venkatd** [comment]
@aj-foster I gave your reader a try:

```elixir
    Call.new(profile, files)
    |> State.new()
    |> Reader.run()
```

And I have to say this codebase is very well built and well documented. Was very easy to pick out just the things I needed to get started.

It's going to save me a lot of time ❤️
